app/src/main/java/com/github/se/travelpouch/ui/dashboard/EditActivity.kt

```diff
@@ -0,0 +1,158 @@
+                    .padding(16.dp)
+                    .padding(paddingValues)
+                    .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
+              OutlinedTextField(
+                  value = title,
+                  onValueChange = { title = it },
+                  enabled = true,
+                  label = { Text("Title") },
+                  modifier = Modifier.testTag("titleField"))
+
+              OutlinedTextField(
+                  value = description,
+                  onValueChange = { description = it },
+                  enabled = true,
+                  label = { Text("Description") },
+                  modifier = Modifier.testTag("descriptionField"))
+
+              OutlinedTextField(
+                  value = location,
+                  onValueChange = {},
+                  enabled = true,
+                  label = { Text("Location") },
+                  modifier = Modifier.testTag("locationField"))
+
+              OutlinedTextField(
+                  value = date,
+                  onValueChange = {
+                    if (it.isDigitsOnly() && it.length <= 8) {
+                      date = it
+                    }
+                  },
+                  enabled = true,
+                  label = { Text("Date") },
+                  visualTransformation = DateVisualTransformation(),
+                  keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                  modifier = Modifier.testTag("dateField"))
+
+              Button(
+                  enabled =
+                      location.isNotBlank() &&
+                          title.isNotBlank() &&
+                          description.isNotBlank() &&
+                          date.isNotBlank(),
+                  onClick = {
+                    try {
+                      val finalDate = convertStringToDate(date, dateFormat, gregorianCalendar)
+
+                      val activity =
+                          Activity(
+                              selectedActivity.value!!.uid,
+                              title,
+                              description,
+                              selectedActivity.value!!.location,
+                              finalDate,
+                              mapOf())
+
+                      activityViewModel.updateActivity(activity, context)
+                    } catch (e: java.text.ParseException) {
+                      Toast.makeText(
+                              context,
+                              "Invalid format, date must be DD/MM/YYYY.",
+                              Toast.LENGTH_SHORT)
+                          .show()
+                    }
+                    navigationActions.navigateTo(Screen.TRAVEL_ACTIVITIES)
+                  },
+                  modifier = Modifier.testTag("saveButton")) {
+                    Text("Save")
+                  }
+            }
+      }
+}
+
```

Comment:
You could add some cursory documentation that way it looks all nice and shiny when you hover over it.

```suggestion
/**
 * Converts a `Date` object to a string in the format "ddMMyyyy".
 *
 * @param date The `Date` object to be converted.
 * @return A string representing the date in the format "ddMMyyyy".
 */
```